### ISSUE TITLE
Add Exceeding vertical boundaries in Validation

### ISSUE THREAD

**Issue by yg8ijvjvjv** [fontforge/fontforge#4550]
The vertical/fontheight boundaries (unsigned Win Ascent and unsigned Win Descent) determine where vertically each line starts and ends. In any usable renderer, vertical boundaries are clipped because on the fundamental level it is impossible for a scanline to be a part of multiple lines.

If Win Ascent and Win Descent are not Is Offset (I recommend it shouldn't be Is Offset by default because new type designers risk introducing optically incorrect vertical spacing and metric incompatibility between versions) the Validation should warn about exceeding vertical boundaries, and the default vertical bounding box check vales in Find Problems should be the font height boundaries.

- [ ] Win Ascent and Win Descent are no longer Is Offset by default
- [ ] Validation warns font height boundaries if not Is Offset
- [ ] Find Problems defaults bounding box to font height boundaries if not Is Offset

**Comment by ctrlcctrlv** [comment]
I agree about Is Offset. I'm not sure I understand the scanline problem.

**Comment by yg8ijvjvjv** [comment]
The scanline concept is used in font rendering, it means each scanline belongs to exactly one text line. The implication of this is that content beyond vertical boundaries (Win Ascent and Win Descent, rounded to integer pixels and half pixels rounded away from baseline) gets clipped and therefore should be warned.

**Comment by ctrlcctrlv** [comment]
Got it. This is two issues. I think you should make a new issue for the Is Offset bug. I will rename your bug to make it clearer to others.

**Comment by yg8ijvjvjv** [comment]
So, if the font exceeds vertical boundaries, the validation might look like this:
![](https://i.imgur.com/CP1rzlA.png)
